### DISCUSSION TITLE
[ZK Stack] - <What is the proper practice for backup my hyperchain data and configuration>

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#613] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Wondering how to back up my hyperchain in case of accident, I can restore the whole chain working.
Im using commit in MAY, sha ea0956f095b4558d94a2b7589156fe4a0754526e.

### Expected Behavior

backup my current data and configuration and hope to restore the status when bad things happen.

### Current Behavior

dont know how to backup

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
Thank you for your question! I’m unable to provide an answer at this moment, but I have forwarded your inquiry to our team. Someone will be with you shortly to assist further. Your patience is appreciated!

**Comment by zk-Lumi** [comment, marked as answer]
Hi @qiwenzhou, thanks for reaching out and that is a great question.

For backing up the chain you should run ->
- DB backups - dumps + point in time backups
- Tree and state keeper storage disk snapshots

How you would do this depends a bit on your setup.

Hope that helps!
